design-proposals/klip-3-serialization-of-single-fields.md

```diff
@@ -0,0 +1,1076 @@
+or :ref:`CREATE STREAM AS SELECT <create-stream-as-select>` statements.
+
+.. note:: This setting has no effect on formats that do not support some 
+          kind of outer record or object. For example, ``DELIMITED``.
+
+.. _ksql-persistence-wrap-single-values:
+
+-----------------------------------
+ksql.persistence.wrap.single.values
+-----------------------------------
+
+Sets the default value for the ``WRAP_SINGLE_VALUE`` property if one is
+not supplied explicitly in :ref:`CREATE TABLE <create-table>`,
+:ref:`CREATE STREAM <create-stream>`, :ref:`CREATE TABLE <create-table-as-select>`
+or :ref:`CREATE STREAM AS SELECT <create-stream-as-select>` statements.
+
+This setting can be toggled using the `SET` command
+
+.. code:: sql
+
+    SET 'ksql.persistence.wrap.single.values'='false';
+
+For more information, refer to the :ref:`CREATE TABLE <create-table>`,
+:ref:`CREATE STREAM <create-stream>`, :ref:`CREATE TABLE <create-table-as-select>`
+or :ref:`CREATE STREAM AS SELECT <create-stream-as-select>` statements.
+
+.. note:: This setting has no effect on formats that do not support some 
+          kind of outer record or object. For example, ``DELIMITED``.
+```
+
+* The `CREATE TABLE` and `CREATE STREAM` sections in `syntax-reference.rst`
+  will have an updated properties section that includes the following rows:
+
+```rst
+ +=========================+========================================================================================================+
+ | WRAP_SINGLE_KEY         | Controls how keys are deserialized where the key's schema contains only a single field.                |
+ |                         |                                                                                                        |
+ |                         | The setting controls how KSQL will deserialize the key of the records in the supplied ``KAFKA_TOPIC``. |
+ |                         | If set to ``true`` KSQL expects the field(s) to have been serialized as named field(s) within a record.|
+ |                         | If set to ``false`` and the key has a single-field schema, KSQL expects the field to have been         |
+ |                         | serialized as an anonymous value.                                                                      |
+ |                         | Setting to ``false`` when the key is a multi-field schema will result in an error                      |
+ |                         |                                                                                                        |
+ |                         | If not supplied, the system default, defined by :ref:`ksql-persistence-wrap-single-keys` and           |
+ |                         | defaulting to ``false``, is used.                                                                      |                                                                                                      |
```

Comment:
```suggestion
 |                         | defaulting to ``true``, is used.                                                                      |                                                                                                      |
```

Reply:
I don't think this is a typo. It's mentioned above that `ksql.persistence.wrap.single.keys` defaults to false (in contrast to `ksql.persistence.wrap.single.values` which defaults to true).

Reply:
Vic is right.  I've updated the doc so that keys default to unwrapped and values to wrapped.  Thinking behind this is detailed in the 'public APIs' section of the KLIP.